modules/core/src/main/scala/com.snowplowanalytics.snowplow.bigquery/Metrics.scala

```diff
@@ -35,20 +36,22 @@ object Metrics {
   private case class State(
     good: Long,
     bad: Long,
-    latency: FiniteDuration
+    latency: FiniteDuration,
+    e2eLatency: FiniteDuration
   ) extends CommonMetrics.State {
     def toKVMetrics: List[CommonMetrics.KVMetric] =
       List(
         KVMetric.CountGood(good),
         KVMetric.CountBad(bad),
-        KVMetric.Latency(latency)
+        KVMetric.Latency(latency),
+        KVMetric.E2ELatency(e2eLatency)
       )
   }
 
   private object State {
     def initialize[F[_]: Functor](sourceAndAck: SourceAndAck[F]): F[State] =
       sourceAndAck.currentStreamLatency.map { latency =>
-        State(0L, 0L, latency.getOrElse(Duration.Zero))
+        State(0L, 0L, latency.getOrElse(Duration.Zero), Duration.Zero)
```

Comment:
We should not emit `Duration.Zero` in circumstances when no events are loaded.  Please be consistent with the lake loader.  The default value should be `None` and we should emit nothing when nothing is loaded.

https://github.com/snowplow-incubator/snowplow-lake-loader/blob/0.6.0/modules/core/src/main/scala/com.snowplowanalytics.snowplow.lakes/Metrics.scala#L46